gobblin-service/src/main/java/org/apache/gobblin/service/modules/flowgraph/Dag.java

```diff
@@ -115,6 +114,36 @@ public boolean isEmpty() {
       }
       this.endNodes = other.endNodes;
     }
+    //Append all the entries from the other dag's parentChildMap to this dag's parentChildMap
```

Comment:
By the way, for line 112, why Dag1's endNodes need to link with Dag2's startNodes in a N x M manner? Do we assume the dependency is either or all?

Reply:
This is because we do not want any of Dag2's nodes to start until all of Dag1's nodes have finished.